src/test/resources/GetClassLoader.java

```diff
@@ -0,0 +1,26 @@
+import javax.ejb.embeddable.EJBContainer;
+
+public class GetClassLoader {
+
+	ClassLoader d = this.getClass().getClassLoader(); // Noncompliant
+
+	public void case1() {
+		ClassLoader c = this.getClass().getClassLoader(); // Noncompliant
+	}
+
+	public void case2() throws ClassNotFoundException{
+		Dummy.class.getClassLoader().loadClass("anotherclass"); // Noncompliant
+	}
+
+	abstract class InnerClass {
+		ClassLoader f = this.getClass().getClassLoader(); // Noncompliant
+
+		public void hello() {
+			this.getClass().getClassLoader(); // Noncompliant
+		}
+	}
+
+	public void usingEJB() {
+		EJBContainer container = new EJBContainer();
+	}
```

Comment:
This last method seems irrelevant. What is it?

Reply:
it shows that the code is in JEE context as it uses an item from `javax` :) . If this piece does not exist then the import will not be parsed in Spoon later as it's regarded as unused import.

Reply:
Understood.